command/agent.go

```diff
@@ -719,11 +679,19 @@ func (c *AgentCommand) Run(args []string) int {
 			}
 			ln = inProcListener
 		} else {
-			ln, tlsConf, err = cache.StartListener(lnConfig)
+			lnBundle, err := cache.StartListener(lnConfig)
 			if err != nil {
 				c.UI.Error(fmt.Sprintf("Error starting listener: %v", err))
 				return 1
 			}
+
+			tlsCfg = lnBundle.TLSConfig
+			ln = lnBundle.Listener
+
+			// Track the reload func, so we can reload later if needed.
+			// NOTE: If we plan to reload entire listeners at some point it won't be enough to just
```

Comment:
I'm personally not convinced about comments like this, and I think we should probably delete these two lines starting with `NOTE`, for a few reasons:
- I doubt we'll actually do the work to reload entire listeners, it seems like a big stretch of what SIGHUP should do, and Vault itself doesn't do it
- If we do this feature, I'm not convinced that this comment will be too helpful

If this were a feature we were planning on doing, and doing soon, I'd feel differently. As-is, I feel like these comments will probably grow old forever.

Reply:
It seems the Vault code base already has lots of uses of `// NOTE:` :( 

It's something I've used in the past to indicate something might need a heads up before you embark on further changes.

I don't think I considered planning input on whether I should write the comment or not, given plans often change. The comment might be useful to me/someone next week or in a few months.

All that being said, I can (will) remove it if it helps resolve outstanding issues on the PR. :)

Reply:
I definitely think that this comment doesn't add sufficiently enough to justify being there, given that we don't see this as being a feature we're planning on adding. Lines are precious, especially within the `Run` function!

Just to be clear, I'm not averse to comments, I just think that this one isn't, likely to add value, and if it ever does (if we ever do decide to support listener reloading, which I don't think we should), the value is not sufficiently high enough to warrant it. Admittedly though, this is a bit of a nit-pick, but it's something I felt strongly enough to bring up.

This was my only outstanding comment on the PR, so I'll approve it, and I'd prefer to see the comment removed but I won't lose sleep if you decide it's worth keeping!

Reply:
I think we should be careful about discouraging people from sharing knowledge and insight in the form of comments.

Personally, I'm in favor of adding `// NOTE` comments like this. They usually reflect hard earned knowledge that the author has only gained after staring and fiddling with said code for hours or days. They're designed specifically to save time and effort for future endeavors. They're typically too brief and/or not actionable enough to warrant a full Jira style write up (which would likely fall to the bottom of the backlog anyway), so if they're not noted in the code, then the learnings are lost, and future developers will likely need to learn these same lessons the hard way, again. And finally, they're in the best possible place, imo, because they're right next to the code they refer to, making their utility much higher than something like a Jira.

From a broad point of view, I'm in favor of almost all comments, as they typically add something, rather than detract. The only comments I'm not in favor of are `// TODO` comments, which typically mean 1) there is work left to be done on the PR or 2) a proper Jira should be filed.

Reply:
I don't disagree that comments are good, but this comment in particular is about a situation that very likely will never come up. In other words, the comment is only useful (and even then, marginally so) if we do a feature we're incredibly unlikely to do (ability to change listeners while Agent is running). In other words, I feel like the likelihood of this helping a future developer is quite low indeed.

In either case, it's like I've said, it's definitely not a hill I'm willing to die on, I just wanted to explain my perspective around this comment in a bit more detail.

Reply:
Thanks for the additional perspective!

> I don't disagree that comments are good, but I was worried that this comment in particular is about a situation that I believe will very likely never come up.

I would caution against this type of thinking. Some of our product direction comes from within HC but some also comes from our customers. I've learned the hard way to never underestimate customers 😆 .

Even if you're right though, and we never implemented this particular feature, I'm not sure I understand why having the comment exist in the code is a bad thing, worthy of intentional removal. At best, it provides useful context to someone else coming to this code in the future. At worst, it's something that most people ignore. But either way, having it in the code isn't actively harmful, as far as I can tell.

In the end, what I'm really trying to do is promote a team culture where people feel comfortable leaving their thoughts and insights in the code, in the form of comments. Vault's code has significantly more comments than any other project I've worked on, and I've derived significant value from them. I would hate to think that kind of practice might stop.